planetscale/imports.go

```diff
@@ -0,0 +1,306 @@
+package planetscale
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type DataImportSource struct {
+	HostName            string `json:"hostname"`
+	Database            string `json:"schema_name"`
+	Port                int    `json:"port"`
+	SSLMode             string `json:"ssl_mode"`
+	SSLVerificationMode ExternalDataSourceSSLVerificationMode
+	UserName            string `json:"username"`
+	Password            string `json:"password"`
+}
+
+type ExternalDataSourceSSLVerificationMode int
+
+const (
+	SSLModeDisabled ExternalDataSourceSSLVerificationMode = iota
+	SSLModePreferred
+	SSLModeRequired
+	SSLModeVerifyCA
+	SSLModeVerifyIdentity
+)
+
+func (sm ExternalDataSourceSSLVerificationMode) String() string {
+	switch sm {
+	case SSLModeDisabled:
+		return "disabled"
+	case SSLModePreferred:
+		return "preferred"
+	case SSLModeRequired:
+		return "required"
+	case SSLModeVerifyCA:
+		return "verify_ca"
+	default:
+		return "verify_identity"
+	}
+}
+
+type DataImportState int
+
+const (
+	DataImportPreparingDataCopy DataImportState = iota
+	DataImportPreparingDataCopyFailed
+	DataImportCopyingData
+	DataImportCopyingDataFailed
+	DataImportSwitchTrafficPending
+	DataImportSwitchTrafficRunning
+	DataImportSwitchTrafficCompleted
+	DataImportSwitchTrafficError
+	DataImportReverseTrafficRunning
+	DataImportReverseTrafficCompleted
+	DataImportReverseTrafficError
+	DataImportDetachExternalDatabaseRunning
+	DataImportDetachExternalDatabaseError
+	DataImportReady
+)
+
+var stateToImportStateMap = map[string]DataImportState{
+	"prepare_data_copy_pending":        DataImportPreparingDataCopy,
```

Comment:
Should we rename `DataImportPreparingDataCopy` to `DataImportPreparingDataCopyPending` ? Looks like we have other variables that don't have the `Pending` suffix.

Reply:
Yeah, the `Pending` suffix only makes sense in the workflow since we know what the next step would be. 
In case of clients, I'd prefer to show the current state.